pkg/i2gw/provider.go

```diff
@@ -55,13 +55,11 @@ type Provider interface {
 
 type CustomResourceReader interface {
 
-	// ReadResourcesFromCluster reads custom resources associated with
-	// the underlying Provider implementation from the kubernetes cluster.
-	ReadResourcesFromCluster(ctx context.Context, customResources interface{}) error
+	// ReadResourcesFromCluster reads custom resources from the kubernetes cluster.
```

Comment:
why removing `associated with the underlying Provider implementation from the kubernetes cluster` ?